joint_trajectory_controller/doc/userdoc.rst

```diff
@@ -101,6 +113,105 @@ A yaml file for using it could be:
               goal: 0.03
 
 
+.. _ROS 2 interface:
+
+ROS 2 interface of the controller [#f1]_
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+There are two mechanisms for sending trajectories to the controller:
+
+~/follow_joint_trajectory (action server) [control_msgs::action::FollowJointTrajectory]
+  Action server for commanding the controller.
+
+~/joint_trajectory (input topic) [trajectory_msgs::msg::JointTrajectory]
+  Topic for commanding the controller.
+
+Both use the ``trajectory_msgs/JointTrajectory`` message to specify trajectories, and require specifying values for all the controller joints (as opposed to only a subset) if ``allow_partial_joints_goal`` is not set to ``True``.
+
+The primary way to send trajectories is through the action interface, and should be favored when execution monitoring is desired.
+Action goals allow to specify not only the trajectory to execute, but also (optionally) path and goal tolerances.
+When no tolerances are specified, the defaults given in the parameter interface are used (see :ref:`parameters`).
+If tolerances are violated during trajectory execution, the action goal is aborted, the client is notified, and the current position is held.
```

Comment:
I think this changed from ROS 1, is the described behavior correct (and intentional?)

Reply:
It should be. Maybe we broke it, but we want to have this behavior

Reply:
I'll investigate this one, maybe I just did not understand the nested if-else clauses.

Reply:
I checked this one now -> it aborts the action.
> Aborted due to state tolerance violation

You meant that the abort is the desired behavior? And not the ROS 1 behavior, where client is only notified but the action is still active?

Reply:
Yes I think we should abort then